ouroboros-consensus-cardano/src/ouroboros-consensus-cardano/Ouroboros/Consensus/Cardano/CanHardFork.hs

```diff
@@ -272,28 +280,28 @@ type CardanoHardForkConstraints c =
 
 instance CardanoHardForkConstraints c => CanHardFork (CardanoEras c) where
   hardForkEraTranslation = EraTranslation {
-      translateLedgerState   =
-          PCons translateLedgerStateByronToShelleyWrapper
-        $ PCons translateLedgerStateShelleyToAllegraWrapper
-        $ PCons translateLedgerStateAllegraToMaryWrapper
-        $ PCons translateLedgerStateMaryToAlonzoWrapper
-        $ PCons translateLedgerStateAlonzoToBabbageWrapper
-        $ PCons translateLedgerStateBabbageToConwayWrapper
+      crossEraTickLedgerState =
+          PCons crossEraTickLedgerStateByronToShelley
+        $ PCons crossEraTickLedgerStateAcrossTPraos
+        $ PCons crossEraTickLedgerStateAcrossTPraos
+        $ PCons crossEraTickLedgerStateAcrossTPraos
+        $ PCons crossEraTickLedgerStateTPraosToPraos   -- TODO is there an Alonzo->Babbage bug wrt extraEntropy?
+        $ PCons crossEraTickLedgerStateAcrossPraos   -- TODO fix Babbage->Conway bug
```

Comment:
Why do the `crossEraTickLedgerState*` functions mention the `ConsensusProtocol`s TPraos and Praos? It would make sense for the `crossEraTickChainDepState` functions, but I think the cross-era ledger state ticking should be orthogonal to what `ConsensusProtocol` is used, right?

I pushed 2713ab6c28b809f77258babd48a2f05d2e493cb6 unifying these.

Reply:
The cast is only needed when the protocol differs, and I considered the concrete types easier to read than the constraints on the polymorphic one.

I was aware that you had unified them on your other PR. I only have a minor preference for the un-unified approach.